src/helper/wrapper/playwrightWrappers.ts

```diff
@@ -56,7 +56,7 @@ export default class PlaywrightWrapper {
     };
 
     const startTime = Date.now();
-    while (Date.now() - startTime < 15000) {
+    while (Date.now() - startTime < 30000) {
       if (await isTextMatch()) {
         return true;
       }
```

Comment:
You're invoking `isTextMatch` asynchronously every 1 ms until 30 seconds of timeout. Maybe, you don't need to check text changes every 1 ms? Instead, you could try checking if checking has changed every 100 ms or every second.